models/vrp.rb

```diff
@@ -407,15 +407,26 @@ def self.deduce_unavailable_days(hash, element, start_index, end_index, type)
       collect_unavaible_day_indices(element, start_index, end_index)
     end
 
-    def self.generate_schedule_indices_from_date(hash)
-      return hash if !hash[:configuration] || !hash[:configuration][:schedule]
-
-      if hash[:configuration][:schedule][:range_indices] &&
-         (hash[:services]&.any?{ |s| s[:unavailable_date_ranges] || s[:unavailable_visit_day_date] })
+    def self.detect_date_indices_inconsistency(hash)
```

Comment:
Lets make this function handle its corner case so that there is no future risk of `undefined method [] for nil:NilClass`

```suggestion
    def self.detect_date_indices_inconsistency(hash)
      return hash unless hash[:configuration] && hash[:configuration][:schedule]

```

Reply:
I agree, even though this check is done in _generate_schedule_indices_from_date_ which is the only function that calls _detect_date_indices_inconsistency_. Should we check this twice ?

Reply:
I think it is better if each function handles its corner case on its own. this fuction cannot know if the caller already checked or not.